### PR TITLE
[webnn] support tests with constant operands

### LINKED ISSUE #38274: webnn/conv2d.https.any.js should add conv2d filter as constant operand
The existing conv2d test cases only define conv2d filter as input operand. 

However, neural networks usually define the filter of conv2d operator as constant operand. And some implementation may only support constant filter operand.

With that, it would be helpful for implementers that add test cases that define conv2d filter as constant operand.

/cc @BruceDai @Honry

### PR DESCRIPTION
**Pull request by BruceDai**
This PR is to fix https://github.com/web-platform-tests/wpt/issues/38274

- update previous conv2d tests with **constant** **filter** and add a test of non-constant **filter**.
- update previous batchNormalization tests with constant **mean** and **variance**, add a test of non-constant **mean** and **variance**

@Honry @huningxin PTAL, thanks.

### CODE REVIEW

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Review by huningxin — APPROVED** [review]
LGTM, thanks!
